browser/ui/webui/settings/brave_settings_leo_assistant_handler.cc

```diff
@@ -41,23 +85,57 @@ void BraveLeoAssistantHandler::OnItemAdded(const sidebar::SidebarItem& item,
                                            size_t index) {
   if (item.built_in_item_type ==
       sidebar::SidebarItem::BuiltInItemType::kChatUI) {
-    NotifyChatUiChanged();
+    NotifyChatUiChanged(true);
   }
 }
 
 void BraveLeoAssistantHandler::OnItemRemoved(const sidebar::SidebarItem& item,
                                              size_t index) {
   if (item.built_in_item_type ==
       sidebar::SidebarItem::BuiltInItemType::kChatUI) {
-    NotifyChatUiChanged();
+    NotifyChatUiChanged(false);
   }
 }
 
-void BraveLeoAssistantHandler::NotifyChatUiChanged() {
+void BraveLeoAssistantHandler::NotifyChatUiChanged(const bool& isLeoVisible) {
   if (!IsJavascriptAllowed()) {
     return;
   }
-  FireWebUIListener("settings-brave-leo-assistant-changed");
+  FireWebUIListener("settings-brave-leo-assistant-changed", isLeoVisible);
+}
+
+void BraveLeoAssistantHandler::HandleToggleLeoIcon(
+  const base::Value::List& args) {
+  auto* service = sidebar::SidebarServiceFactory::GetForProfile(profile_);  
+
+  if(!ShowLeoAssistantIconVisibleIfNot(service)) {
+    HideLeoAssistantIconIfNot(service);
+  }
+}
+
+void BraveLeoAssistantHandler::HandleGetLeoIconVisibility(
+  const base::Value::List& args) {
+  auto* service = sidebar::SidebarServiceFactory::GetForProfile(profile_);
+  const auto hidden_items = service->GetHiddenDefaultSidebarItems();
+  AllowJavascript();
+  ResolveJavascriptCallback(args[0],
+    !base::ranges::any_of(hidden_items, [](const auto& item) {
+        return item.built_in_item_type ==
+               sidebar::SidebarItem::BuiltInItemType::kChatUI;
+      }));
+}
+
+void BraveLeoAssistantHandler::HandleResetLeoData(const base::Value::List& args) {
+  auto* service = sidebar::SidebarServiceFactory::GetForProfile(profile_);
+
+  ShowLeoAssistantIconVisibleIfNot(service);
+  profile_->GetPrefs()->SetBoolean(ai_chat::prefs::kBraveChatHasSeenDisclaimer,
+                                  false);
+  profile_->GetPrefs()->SetBoolean(
+      ai_chat::prefs::kBraveChatAutoGenerateQuestions, false);
+  
+  AllowJavascript();
+  ResolveJavascriptCallback(args[0], true);
```

Comment:
I guess we only pass true in here because we need to return something? Out of curiosity, can we pass in `nullptr` or base::Value() here instead? (I don't know if it's possible/better, so I'll leave it up to you).

Reply:
Removed the line `ResolveJavascriptCallback(args[0], true);`
as the return value for `resetLeoData`  is void